test/es-module/test-esm-loader-globalpreload-hook.mjs

```diff
@@ -0,0 +1,145 @@
+      '--no-warnings',
+      '--experimental-loader',
+      'data:text/javascript,export function globalPreload(){return `console.log(getBuiltin("os").arch())`}',
+      fixtures.path('empty.js'),
+    ]);
+
+    assert.strictEqual(stderr, '');
+    assert.strictEqual(stdout.trim(), os.arch());
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should throw when loading node:test without node: prefix', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      'data:text/javascript,export function globalPreload(){return `getBuiltin("test")()`}',
+      fixtures.path('empty.js'),
+    ]);
+
+    assert.match(stderr, /ERR_UNKNOWN_BUILTIN_MODULE/);
+    assert.strictEqual(stdout, '');
+    assert.strictEqual(code, 1);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should register globals set from globalPreload', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      'data:text/javascript,export function globalPreload(){return "this.myGlobal=4"}',
+      '--print', 'myGlobal',
+    ]);
+
+    assert.strictEqual(stderr, '');
+    assert.strictEqual(stdout.trim(), '4');
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should log console.log calls returned from globalPreload', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      'data:text/javascript,export function globalPreload(){return `console.log("Hello from globalPreload")`}',
+      fixtures.path('empty.js'),
+    ]);
+
+    assert.strictEqual(stderr, '');
+    assert.strictEqual(stdout.trim(), 'Hello from globalPreload');
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should crash if globalPreload returns code that throws', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      'data:text/javascript,export function globalPreload(){return `throw new Error("error from globalPreload")`}',
+      fixtures.path('empty.js'),
+    ]);
+
+    assert.match(stderr, /error from globalPreload/);
+    assert.strictEqual(stdout, '');
+    assert.strictEqual(code, 1);
+    assert.strictEqual(signal, null);
+  });
+
+  it('should have a `this` value that is not bound to the loader instance', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      `data:text/javascript,export function globalPreload() { if (this != null) { throw new Error('hook function must not be bound to ESMLoader instance') } return ''; }`,
```

Comment:
```suggestion
      `data:text/javascript,export ${function globalPreload() { if (this != null) { throw new Error('hook function must not be bound to ESMLoader instance') }} }`,
```

Reply:
The previous version had `return ''` as part of the test. Since `globalPreload` is supposed to return a string, I assume that’s preferable to keep?

Reply:
It's probably from a time where we accidentally removed the support for returning `undefined`. No strong feelings, my suggestion is not about the `return ''`